File: raw/raw-text-services/src/main/java/ir/ac/iust/dml/kg/raw/services/web/filter/FilterRegistrationConfiguration.java
/*
 * Farsi Knowledge Graph Project
 *  Iran University of Science and Technology (Year 2017)
 *  Developed by Majid Asgari.
 */

package ir.ac.iust.dml.kg.raw.services.web.filter;

import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;


@Configuration
public class FilterRegistrationConfiguration {

  @Bean
  public FilterRegistrationBean someFilterRegistration() {
    FilterRegistrationBean registration = new FilterRegistrationBean();
    registration.setFilter(new SimpleCORSFilter());
    registration.addUrlPatterns("/*");
    registration.setName("corsFilter");
    registration.setOrder(1);
    return registration;
  }
}

File: update/update_manager/src/main/java/ir/ac/iust/dml/kg/search/feedback/web/commons/filter/FilterRegistrationConfiguration.kt
/*
 * Farsi Knowledge Graph Project
 * Iran University of Science and Technology (Year 2017)
 * Developed by Majid Asgari.
 */

package ir.ac.iust.dml.kg.search.feedback.web.commons.filter

import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class FilterRegistrationConfiguration {

  @Bean
  open fun someFilterRegistration(): FilterRegistrationBean {
    val registration = FilterRegistrationBean()
    registration.filter = SimpleCORSFilter()
    registration.addUrlPatterns("/*")
    registration.setName("corsFilter")
    registration.order = 1
    return registration
  }
}